btx/processing/rawimagetimetool.py

```diff
@@ -0,0 +1,575 @@
+import subprocess # Use has been deprecated
+import os # Directory creation
+# from btx.misc.shortcuts import fetch_latest # Retrieving latest calibration
+import glob
+
+def fetch_latest(fnames, run):
+    """! Fetch the most recently created (in terms of run numbers) file.
+    Here we assume that files are named /{base_path}/r{run:04}.* .
+    
+    Parameters
+    ----------
+    @param fnames (str) glob-expandable string pointing to geom or mask files
+    @param run (int) run number
+    
+    Returns
+    -------
+    @return fname (str) filename of relevant geom or mask file
+    """
+    fnames = glob.glob(fnames)
+    avail = [os.path.basename(f)[1:].split('.')[0] for f in fnames]
+    avail = np.array([int(a) for a in avail])
+    sort_idx = np.argsort(avail)
+    idx = np.searchsorted(avail[sort_idx], run, side='right') - 1
+    return fnames[sort_idx[idx]]
+    try:
+        return fnames[sort_idx[idx]]
+    except IndexError:
+        print('File not found.')
+        return ''
+
+class RawImageTimeTool:
+    """! Class to reimplement time tool analysis at LCLS from raw images.
+
+    Uses psana to interface with experimental data to retrieve raw camera images.
+    Edge detection is implemented in order to perform jitter correction.
+
+    Properties:
+    -----------
+    model - Retrieve or load polynomial model coefficients.
+
+    Methods:
+    --------
+    __init__(self, expmt: str, savedir: str) - Instantiate an analysis object for experiment expmt.
+    open_run(self, run: str) - Open DataSource for a specified run.
+    get_tt_name(self) - Determine time tool camera name in data files.
+    calibrate(self, run: str, order: int, figs=True) - Calibrate the analysis object for jitter correction on specified run.
+    process_run(self) - Perform edge detection on all events in a run.
+    detect_edge(self, img: np.array, kernel: np.array) - Perform edge detection on an image.
+    crop_image(self, img: np.array) - Crop time tool image to ROI.
+    fit_calib(self, delays: np.array, edges: np.array, amplitudes: np.array,
+            fwhm: np.array = None, order: int = 2) - Fit polynomial to calibration run data.
+    ttstage_code(self, hutch) - Return the time tool target time stage for a given hutch.(
+    edge_to_time(self, edge)
+    actual_time(self, edge, nominal)
+    timetool_correct(self, edge, nominal, model, figs=True)
+    plot_calib
+    plot_hist
+    """
+    # Class vars
+    ############################################################################
+
+    # Methods
+    ############################################################################
+
+    def __init__(self, expmt: str, savedir: str):
+        """! Initializer for the time tool analysis object.
+
+        @param expmt (str) Experiment accession name. E.g. cxilr1234.
+        @param savedir (str) Directory where any output files are saved.
+        """
+        ## @var expmt
+        # (str) Experiment accession name
```

Comment:
Doxygen generates explanations for these variables. An interested user can inspect the documentation for these instance variables. They wouldn't otherwise appear because they aren't parameters or return values.